prclz/reblock/reblock.py

```diff
@@ -0,0 +1,205 @@
+import argparse
+import os
+import time
+from pathlib import Path
+from typing import List, Tuple, Callable
+import geopandas as gpd
+import igraph
+import numpy as np
+import pandas as pd
+import tqdm
+from shapely.ops import polygonize
+from shapely.wkt import dumps
+from shapely.geometry import (LinearRing, LineString, MultiLineString,
+                              MultiPoint, MultiPolygon, Point, Polygon)
+from logging import basicConfig, debug, info, warning, error
+
+from .planar_graph import PlanarGraph
+
+def block_to_gadm(block: str) -> str:
+    """Just grabs the GADM from a block"""
+    block_rev = block[::-1]
+    idx = block_rev.index("_") + 1
+    return block[0:-idx]
+
+def add_buildings(graph: PlanarGraph, 
+                  bldg_centroids: List[Point],
+                  ) -> PlanarGraph:
+    """Add bldg centroids to planar graph"""
+
+    total_blgds = len(bldg_centroids)
+    for i, bldg_node in enumerate(bldg_centroids):
+        graph.add_node_to_closest_edge(bldg_node, terminal=True)
+
+    if total_blgds > 0:
+        graph._cleanup_linestring_attr()
+    return graph 
+
+def clean_graph(graph: PlanarGraph) -> PlanarGraph:
+    """
+    Some graphs are malformed bc of bad geometries and 
+    we take the largest connected subgraph
+    """
+    is_conn = graph.is_connected()
+    if is_conn:
+        info("Graph is connected...")
+        return graph, 1
+    else:
+        info("Graph is NOT connected...")
+        components = graph.components(mode=igraph.WEAK)
+        num_components = len(components)
+        comp_sizes = [len(idxs) for idxs in components]
+        arg_max = np.argmax(comp_sizes)
+        comp_indices = components[arg_max]
+
+        debug("len comp_indices = {}".format(len(comp_indices)))
```

Comment:
a) turn log messages into full sentences

b) prefer old-school string interpolation for log messages (see this discussion [here](https://reinout.vanrees.org/weblog/2015/06/05/logging-formatting.html))

so the debug call would look like:
```debug("Number of component indices = %s", len(comp_indices))```

Reply:
interesting! Done